appcelerator/titanium_mobile
# fix(android): remove usage of getResourceAsStream()

[JIRA Ticket](https://jira.appcelerator.org/browse/TIMOB-26910)

**Summary:**
- Remove usage of `getResourceAsStream()` as this method has high overhead
- Utilize `getResource()` or `getAsset()` where appropriate

---
**Test 1:**
_(This verifies we can still read our "build.properties" file.)_
1. Create a default Classic Titanium app.
2. Build and run on Android.
3. Look for a log message like the below on startup. Verify that the "Powered by Titanium" version is `8.3.0` and not `1.0.0`
`<YourAppName> <YourAppVersion> (Powered by Titanium 8.3.0.2d9d45bfc7)`

---
**Test 2:**
_(This verifies that our WebView JS bindings still work.)_
1. Build and run the below code on Android.
2. Wait for the `WebView` to load the appcelerator website.
3. Verify the following gets logged.
`@@@ document.title: Home - Appcelerator | The Mobile First Platform`

```javascript
var lastLoadedTitle = null;
var window = Ti.UI.createWindow();
var webView = Ti.UI.createWebView({
	url: "https://appcelerator.com",
});
webView.addEventListener("load", function(e) {
	var loadedTitle = webView.evalJS("document.title;");
	if (loadedTitle !== lastLoadedTitle) {
		lastLoadedTitle = loadedTitle;
		Ti.API.info("@@@ document.title: " + loadedTitle);
	}
});
window.add(webView);
window.open();
```

---
**Test 3:**
_(This verifies that our WebView JS bindings still work.)_
1. Build and run the below code on Android.
2. Verify that you see a WebView which counts-down from 5 seconds.
3. Verify that a `DOMCountentLoaded` toast message appears at the bottom of the screen when the countdown reaches zero.

```javascript
var htmlText =
		'<!DOCTYPE html>' +
		'<html>' +
		'	<head>' +
		'		<meta name="viewport" content="width=device-width, initial-scale=1.0">' +
		'	</head>' +
		'	<body>' +
		'		<p>HTML JavaScript Interop Test</p>' +
		'		<p id="label"></p>' +
		'	</body>' +
		'	<script>' +
		'		document.addEventListener("DOMContentLoaded", function(e) {' +
		'			Ti.API.info("@@@ Firing events from HTML");' +
		'			Ti.App.fireEvent("app:htmlLoaded", { isRequestingStartTimer: true });' +
		'			Ti.App.fireEvent("app:htmlRequestingShowToast", { message: "DOMContentLoaded" });' +
		'		});' +
		'		function updateLabelTo(x) {' +
		'			document.getElementById("label").innerHTML = x;' +
		'		}' +
		'	</script>' +
		'</html>';

var window = Ti.UI.createWindow();
var webView = Ti.UI.createWebView({
	html: htmlText,
});
window.add(webView);
window.open();

var timerId;
var timerCount;
function updateWebView() {
	timerCount = 5;
	function onUpdate() {
		if (timerCount > 0) {
			Ti.API.info("@@@ Calling evalJS()");
			webView.evalJS("updateLabelTo('Reload in:  " + timerCount + "')", function() {});
			timerCount--;
		} else {
			Ti.API.info("@@@ Reloading WebView");
			clearInterval(timerId);
			timerId = null;
			timerCount = 5;
			webView.html = htmlText;
		}
	}
	timerId = setInterval(function(e) {
		onUpdate();
	}, 1000);
	onUpdate();
}

Ti.App.addEventListener("app:htmlLoaded", function(e) {
	Ti.API.info("@@@ Received event 'htmlLoaded'");
	if (e.isRequestingStartTimer) {
		updateWebView();
	}
});

Ti.App.addEventListener("app:htmlRequestingShowToast", function(e) {
	if (Ti.App.Android || Ti.App.Windows) {
		if (e.message) {
			Ti.UI.createNotification({
				message: e.message,
				duration: Ti.UI.NOTIFICATION_DURATION_SHORT,
			}).show();
		}
	}
});
```
***Test 4***
_(This verifies our TableView icons still work.)_
```JS
const win = Ti.UI.createWindow({
    backgroundColor: 'gray'
});
const table = Ti.UI.createTableView({
    data: [
        {
            title: 'Apples',
            hasCheck: true
        },
        {
            title: 'Bananas',
            hasChild: true
        },
        {
            title: 'Carrots'
        },
        {
            title: 'Potatoes'
        }
    ]
});
win.add(table);
win.open();
```
